### PR TITLE
Isaicp 3254: Implement a contact form

### PR DESCRIPTION
**Pull request by pfrenssen**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
         "drupal/core": "8.3.x-dev#076f41fa03bca938b4ccbed21c0289a514336863",
         "drupal/digital_size_formatter": "~1.0-alpha1",
         "drupal/ds": "^3.0",
-        "drupal/facets": "~1.0@alpha",
+        "drupal/facets": "1.0-alpha7",
```
Comment (pfrenssen) [inline composer.json]: This is temporary, it will be fixed in ISAICP-3246.

File: tests/src/Context/DrupalContext.php [hunk 2]
```diff
@@ -30,37 +30,13 @@ public function loggedIn() {
...
-      if (!isset($user->pass)) {
-        $user->pass = $this->getRandom()->name();
-      }
-      $this->userCreate($user);
-
-      // Assign roles.
-      foreach ($roles as $role) {
-        $this->getDriver()->userAddRole($user, $role);
-      }
+      $this->createUser($userHash);
```
Comment (pfrenssen) [inline tests/src/Context/DrupalContext.php]: I factored this out so we can reuse the support that this has for user roles in the step that creates a single user. This now also completely aligns the two steps (create a single user / create multiple users) so they work in exactly the same way.

File: web/modules/custom/joinup_notification/joinup_notification.behat.inc [hunk 3]
```diff
@@ -32,63 +32,89 @@ class JoinupNotificationSubContext extends DrupalSubContextBase implements Drupa
...
-    $values = $table->getRowsHash();
-    $recipient = $values['recipient'];
-    $message_template_id = $values['template'];
-    $subject = $values['subject'];
-    $body = $values['body'];
+    /** @var string $body */
+    /** @var string $recipient */
+    /** @var string $subject */
+    /** @var string $template */
+    extract($table->getRowsHash());
```
Comment (pfrenssen) [inline web/modules/custom/joinup_notification/joinup_notification.behat.inc]: :pancakes:

File: web/profiles/joinup/joinup.info.yml [hunk 4]
```diff
@@ -100,18 +104,16 @@ config_devel:
   - core.date_format.html_week
   - core.date_format.html_year
   - core.date_format.html_yearless_date
-  - core.date_format.date_only
```
Comment (pfrenssen) [inline web/profiles/joinup/joinup.info.yml]: Alphabetical ordering mess :stuck_out_tongue:

File: web/modules/custom/contact_form/config/install/message.template.contact_form_submission.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+langcode: en
+status: true
+dependencies: {  }
+template: contact_form_submission
+label: 'Contact form submission'
+description: 'Message to the moderators when the contact form is submitted.'
+text:
+  -
+    value: "<p>Joinup - Contact form submission</p>\r\n"
```
Comment (idimopoulos) [inline web/modules/custom/contact_form/config/install/message.template.contact_form_submission.yml]: \r\n?? are you using windows?? :P

File: web/modules/custom/contact_form/src/Controller/ContactFormController.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ * @package Drupal\contact_form\Controller
+ */
+class ContactFormController extends ControllerBase {
+
+  /**
+   * The notification sender.
+   *
+   * @var \Drupal\joinup_notification\NotificationSenderService
+   */
+  protected $notificationSender;
```
Comment (idimopoulos) [inline web/modules/custom/contact_form/src/Controller/ContactFormController.php]: This is not used. It can be removed.

**Pull request merged**